### PR TITLE
Fixes #16616: rl parser improvements and qol + integration tests

### PR DESCRIPTION
**Pull request by gpoblon**
added utils to improve lisibility and quality of the parser
added various integration tests
modified some parser behaviors
fixed some bugs related to error context

### CODE REVIEW

File: rudder-lang/src/logger.rs [hunk 1]
```diff
@@ -145,11 +145,12 @@ pub fn print_output_closure(
         false => {
             let res_str = match is_success {
                 true => format!(
-                    "Everything worked as expected, \"{}\" generated from \"{}\"",
+
```
Comment (gpoblon) [inline rudder-lang/src/logger.rs]: will be removed

**Comment by gpoblon** [comment]
I don't know what went wrong, anyway tests content is now up to date, all tests pass

**Comment by peckpeck** [comment]
your last commit doesn't contain the fixup! prefix, so it won't be merged properly

**Comment by gpoblon** [comment]
my bad sorry, rebased

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**